### PR TITLE
Contact Form: Show child blocks in the Block Library

### LINKED ISSUE #26884: Child blocks of the form block are not showing in the library or search
If users are searching via the inserter for the term "form", they get a suggestion, of course. 

But what if a user is not thinking about creating a "big form", and therefore also not searching for the term "form"?

Maybe someone is looking for something much simpler, like just a "name input" field, an "email input" field, or just a "text input" field?

None of the search terms above would give any results, even though all these child blocks exist. Not even the form block is listed for these keywords. 

-> We could increase the discoverability of the form block by optimizing the block metadata and even create a few simpler variations of the form block that just include one specific form child-block and a submit button, like for example a simple text input field + submit button. 

These very simple form variations, limited in their functionality, could be though the starting point for users to add more input fields and create more complex forms.

List of current Jetpack Forms child-blocks:
Image

### PR DESCRIPTION
**Pull request by jcheringer**
Fixes #26884

#### Changes proposed in this Pull Request:

Uses the Forms child blocks to create simple variations to allow form fields to be searchable in the Block Library.
Introduces the flag `hiddenFromPicker` to allow some variations to be hidden from the Form block placeholder.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Open the Page/Post editor and click the "+" icon at the top-left corner to open the Block Library
* You should be able to search for any Form child block. E.g. `Text`, `Email`, `URL`, `Name`, etc.
* By selecting any of the blocks, a Form block should be added to the Post/Page containing the selected child block and a submit button

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/contact-form/child-blocks.js [hunk 1]
```diff
@@ -0,0 +1,457 @@
+import { createBlock, getBlockType } from '@wordpress/blocks';
```
Comment (jcheringer) [inline projects/plugins/jetpack/extensions/blocks/contact-form/child-blocks.js:1]: This was just an extraction from `projects/plugins/jetpack/extensions/blocks/contact-form/index.js` to allow the child blocks to be used in the `variations.js`

**Review by ice9js — DISMISSED** [review]
Works for me 👍

**Comment by jcheringer** [comment]
Hey @ice9js, could you give me another 👍 ?
Your approval was dismissed because I rebased the changes.

**Review by ice9js — APPROVED** [review]
Still works :)

**Review by kraftbj — APPROVED** [review]

**Pull request merged**